<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">.lst-kix_z0595nuax9qs-0>li:before{content:"-  "}ul.lst-kix_40tprdp2abg-7{list-style-type:none}.lst-kix_40tprdp2abg-1>li:before{content:"-  "}.lst-kix_40tprdp2abg-2>li:before{content:"-  "}ul.lst-kix_40tprdp2abg-8{list-style-type:none}.lst-kix_40tprdp2abg-0>li:before{content:"-  "}.lst-kix_40tprdp2abg-4>li:before{content:"-  "}.lst-kix_z0595nuax9qs-1>li:before{content:"-  "}.lst-kix_z0595nuax9qs-4>li:before{content:"-  "}ul.lst-kix_40tprdp2abg-0{list-style-type:none}ul.lst-kix_40tprdp2abg-1{list-style-type:none}ul.lst-kix_40tprdp2abg-2{list-style-type:none}.lst-kix_z0595nuax9qs-2>li:before{content:"-  "}ul.lst-kix_40tprdp2abg-3{list-style-type:none}.lst-kix_40tprdp2abg-3>li:before{content:"-  "}ul.lst-kix_40tprdp2abg-4{list-style-type:none}.lst-kix_z0595nuax9qs-3>li:before{content:"-  "}ul.lst-kix_40tprdp2abg-5{list-style-type:none}ul.lst-kix_40tprdp2abg-6{list-style-type:none}ul.lst-kix_z0595nuax9qs-5{list-style-type:none}ul.lst-kix_z0595nuax9qs-6{list-style-type:none}ul.lst-kix_z0595nuax9qs-3{list-style-type:none}ul.lst-kix_z0595nuax9qs-4{list-style-type:none}ul.lst-kix_z0595nuax9qs-1{list-style-type:none}ul.lst-kix_z0595nuax9qs-2{list-style-type:none}.lst-kix_40tprdp2abg-8>li:before{content:"-  "}ul.lst-kix_z0595nuax9qs-0{list-style-type:none}.lst-kix_40tprdp2abg-5>li:before{content:"-  "}.lst-kix_40tprdp2abg-6>li:before{content:"-  "}.lst-kix_40tprdp2abg-7>li:before{content:"-  "}ul.lst-kix_z0595nuax9qs-7{list-style-type:none}ul.lst-kix_z0595nuax9qs-8{list-style-type:none}.lst-kix_z0595nuax9qs-8>li:before{content:"-  "}.lst-kix_z0595nuax9qs-5>li:before{content:"-  "}.lst-kix_z0595nuax9qs-6>li:before{content:"-  "}.lst-kix_z0595nuax9qs-7>li:before{content:"-  "}.lst-kix_lnlbj3dhp3t-4>li:before{content:"-  "}.lst-kix_371x4qwtpdjv-6>li:before{content:"-  "}.lst-kix_371x4qwtpdjv-5>li:before{content:"-  "}.lst-kix_371x4qwtpdjv-7>li:before{content:"-  "}.lst-kix_lnlbj3dhp3t-5>li:before{content:"-  "}.lst-kix_lnlbj3dhp3t-1>li:before{content:"-  "}.lst-kix_371x4qwtpdjv-3>li:before{content:"-  "}.lst-kix_lnlbj3dhp3t-2>li:before{content:"-  "}.lst-kix_371x4qwtpdjv-4>li:before{content:"-  "}.lst-kix_lnlbj3dhp3t-3>li:before{content:"-  "}ul.lst-kix_lnlbj3dhp3t-7{list-style-type:none}ul.lst-kix_lnlbj3dhp3t-8{list-style-type:none}ul.lst-kix_lnlbj3dhp3t-5{list-style-type:none}ul.lst-kix_lnlbj3dhp3t-6{list-style-type:none}.lst-kix_371x4qwtpdjv-2>li:before{content:"-  "}ul.lst-kix_lnlbj3dhp3t-0{list-style-type:none}.lst-kix_371x4qwtpdjv-1>li:before{content:"-  "}.lst-kix_lnlbj3dhp3t-0>li:before{content:"-  "}ul.lst-kix_lnlbj3dhp3t-3{list-style-type:none}.lst-kix_371x4qwtpdjv-0>li:before{content:"-  "}ul.lst-kix_lnlbj3dhp3t-4{list-style-type:none}ul.lst-kix_lnlbj3dhp3t-1{list-style-type:none}ul.lst-kix_lnlbj3dhp3t-2{list-style-type:none}ul.lst-kix_371x4qwtpdjv-6{list-style-type:none}ul.lst-kix_371x4qwtpdjv-7{list-style-type:none}ul.lst-kix_371x4qwtpdjv-4{list-style-type:none}ul.lst-kix_371x4qwtpdjv-5{list-style-type:none}ul.lst-kix_371x4qwtpdjv-8{list-style-type:none}.lst-kix_lnlbj3dhp3t-8>li:before{content:"-  "}.lst-kix_lnlbj3dhp3t-6>li:before{content:"-  "}ul.lst-kix_371x4qwtpdjv-2{list-style-type:none}.lst-kix_371x4qwtpdjv-8>li:before{content:"-  "}ul.lst-kix_371x4qwtpdjv-3{list-style-type:none}.lst-kix_lnlbj3dhp3t-7>li:before{content:"-  "}ul.lst-kix_371x4qwtpdjv-0{list-style-type:none}ul.lst-kix_371x4qwtpdjv-1{list-style-type:none}ol{margin:0;padding:0}table td,table th{padding:0}.c1{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:justify;height:11pt}.c19{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:20pt;font-family:"Arial";font-style:normal}.c0{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial";font-style:normal}.c5{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial"}.c8{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:justify}.c13{color:#000000;text-decoration:underline;vertical-align:baseline;font-size:16pt;font-family:"Arial";font-style:normal}.c10{color:#000000;text-decoration:none;vertical-align:baseline;font-size:16pt;font-family:"Arial";font-style:normal}.c21{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:center}.c14{color:#000000;text-decoration:underline;vertical-align:baseline;font-size:14pt;font-family:"Arial";font-style:normal}.c17{padding-top:20pt;padding-bottom:6pt;line-height:1.15;page-break-after:avoid;text-align:justify}.c18{font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial"}.c22{background-color:#ffffff;max-width:451.4pt;padding:72pt 72pt 72pt 72pt}.c11{margin-left:36pt;padding-left:0pt}.c24{color:#0366d6;font-size:12pt}.c20{font-size:14pt;text-decoration:underline}.c7{padding:0;margin:0}.c15{color:inherit;text-decoration:inherit}.c23{text-indent:36pt}.c3{color:#00b050}.c4{font-weight:700}.c12{color:#0000ff}.c16{color:#990000}.c6{color:#38761d}.c2{font-style:italic}.c9{color:#980000}.title{padding-top:0pt;color:#000000;font-size:26pt;padding-bottom:3pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:0pt;color:#666666;font-size:15pt;padding-bottom:16pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:20pt;color:#000000;font-size:20pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-size:16pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:16pt;color:#434343;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:14pt;color:#666666;font-size:12pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body class="c22"><h1 class="c17" id="h.fr11brr5f70d"><span class="c19">Report Group 5</span></h1><p class="c21"><span class="c4 c13">Ancient HGT to Firm5 (from unknown donor)</span></p><p class="c21"><span class="c4 c10">Joaquim, Virginie &amp; Kamil</span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span class="c0">We are going to investigate which genes of our Firm5 group are potentially acquired by ancient horizontal gene transfer from unknown donor. </span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span>Our GitHub repository is: </span><span class="c24"><a class="c15" href="https://www.google.com/url?q=https://github.com/KamilSJaron/SAGE_Firm5_specific_HGT.git&amp;sa=D&amp;ust=1496749944551000&amp;usg=AFQjCNHuqRq0uVqrenjncIEzdtWkkkbFew">https://github.com/KamilSJaron/SAGE_Firm5_specific_HGT.git</a></span></p><p class="c8"><span class="c0">On Vital-IT, you can access the same repo by typing: </span></p><p class="c8 c23"><span class="c5 c2">cd /scratch/beegfs/monthly/mls_2016/claivaz_ricci/SAGE_Firm5_specific_HGT</span></p><p class="c8"><span>In this folder, README.md comprises all informations about our </span><span class="c9">folders</span><span>&nbsp;and </span><span class="c6">files</span><span>. To make it short, </span><span class="c9">data</span><span>&nbsp;contains all inputs and outputs files whereas </span><span class="c9">scripts</span><span>&nbsp;contains all </span><span class="c12">scripts</span><span class="c0">&nbsp;for data analysis. </span></p><p class="c8"><span class="c9">Folders are in red</span><span>, </span><span class="c6">files (inputs and outputs) in green</span><span>&nbsp;and </span><span class="c12">scripts in blue</span><span class="c0">.</span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span class="c4 c14">Summary:</span></p><p class="c8"><span class="c0">We are going to select all gene families which contains orthologous genes if at least 80% of all strains in a given group of bees are present. Then, we are going to extract each protein sequences of all orthologous genes of each group of bees and blast them against RefSeq database (ncbi database). Then, we are going to parse blast results according to parameters. Thanks to blast results, we are going to determine hierarchical taxonomy and hierarchical taxonomy distance for each blast hit. This will allow us to plot % of identity of blast hits against hierarchical taxonomy distance. In the end, all blast hits corresponding to &lsquo;Lactobacillus&rsquo; (in addition to supposed contaminations) will be discarded as we are interested in HGT from species other than Lactobacillus. From blast results, we will perform 2 different analysis. The first one is constructing phylogenetic trees of gene families potentially acquired by HGT and the second one is to predict genomic islands of gene families potentially acquired by HGT.</span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span class="c14 c4">Group of bees:</span></p><p class="c8"><span class="c4">Bumble_bees:</span><span class="c0">&nbsp;B_pascuorum+B_bohemicus </span></p><p class="c8"><span class="c0">(&#39;F225&#39;,&#39;F230&#39;,&#39;F233&#39;,&#39;F234&#39;,&#39;F236&#39;,&#39;F237&#39;)+(&#39;F228&#39;, &#39;F245&#39;,&#39;F246&#39;,&#39;F247&rsquo;)</span></p><p class="c8"><span class="c4">Honey_bees:</span><span class="c0">&nbsp;A_mellifera </span></p><p class="c8"><span class="c0">(&#39;JF72&#39;,&#39;JF73&#39;,&#39;JG30&#39;,&#39;JF74&#39;,&#39;JF75&#39;,&#39;JF76&#39;,&#39;F259&#39;,&#39;F260&#39;,&#39;F261&#39;,&#39;F262&#39;,&#39;F263&#39;,&#39;L183&#39;,&#39;L184&#39;,&#39;L185&#39;,&#39;L186&#39;)</span></p><p class="c8"><span class="c4">Bumble_Honey_bees:</span><span class="c0">&nbsp;Bumble_bees+Honey_bees</span></p><p class="c8"><span class="c4">Outgroup:</span><span class="c0">&nbsp;other strains which are not Bumble_bees and Honey_bees &nbsp;</span></p><p class="c8"><span class="c0">(&#39;JG29&#39;,&#39;LA14&#39;,&#39;LA2&#39;,&#39;LDB&#39;,&#39;LGAS&#39;,&#39;LHV&#39;,&#39;LJP&#39;,&#39;WANG&#39;)</span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span class="c4 c20">Steps:</span></p><p class="c8"><span>1. For each group of bees (Bumble_bees, Honey_bees, Bumble_Honey_bees, Outgroup), </span><span class="c12">subgroup_sort.py</span><span>&nbsp;looks into each Gene_family_* (each line) of </span><span class="c6">GeneFamilies.txt</span><span class="c0">. If a Gene_family_* line contains at least 80% of all strains present in the group of bees, it stores the Gene_family_* line into its corresponding file. </span></p><p class="c8"><span class="c6 c2">Input: GeneFamilies.txt </span><span class="c18 c2 c9">(in data)</span></p><p class="c8"><span class="c6 c2">Outputs: Bumble_bees.txt, Honey_bees.txt, Bumble_Honey_bees.txt and Outgroup.txt </span><span class="c18 c2 c9">(in data/sort_group)</span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span>2. For each group of bees and for each Gene_family_*, </span><span class="c12">proteinseqextract.py</span><span class="c0">&nbsp;extracts all protein sequences for further &lsquo;Blast&rsquo; step. This function creates a folder for each group of bees which comprises fasta files for each Gene_family_* containing protein sequences. </span></p><p class="c8"><span class="c6 c2">Inputs: outputs of </span><span class="c12 c2">subgroup_sort.py</span><span class="c6 c2">&nbsp;</span><span class="c2">in</span><span class="c2 c9">&nbsp;data/sort_group</span><span class="c6 c2">&nbsp;(Bumble_bees.txt, Honey_bees.txt, Bumble_Honey_bees.txt or Outgroup.txt), all_proteins.fasta</span><span class="c2">&nbsp;</span><span class="c2 c9">(in mls_2016/blast folder)</span><span class="c18 c6 c2">&nbsp;</span></p><p class="c8"><span class="c6 c2">Outputs: </span><span class="c2 c9">folders in data of Bumble_bees_proteins, Honey_bees_proteins, Bumble_Honey_bees_proteins</span><span class="c2">&nbsp;- each folder contains .fasta files of gene families </span><span class="c6 c2">(Gene_family_*.fasta)</span><span class="c5 c2">&nbsp;- a file concerns one gene family and contains all reference_genome|protein_ID and their sequence in fasta format</span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span>3. For every protein sequences present in each </span><span class="c6">Gene_family_*.fasta</span><span>&nbsp;files in each folder of group of bees (</span><span class="c9">data/Bumble_bees_proteins, data/Honey_bees_proteins, data/Bumble_Honey_bees_proteins</span><span>), </span><span class="c12">blast_gene_families.sh</span><span>&nbsp;and </span><span class="c12">function_blast.sh</span><span>&nbsp;perform blastp against RefSeq database (module Blast is previously added on Vital-IT, see </span><span class="c12">blast_gene_families.sh</span><span>). RefSeq database is suitable as it is big (and manually curated) and it will allow us to access taxonomic informations from NCBI for further &lsquo;Phylogeny&rsquo; step. </span><span class="c12">blast_gene_families.sh</span><span>&nbsp;runs blastp command specifying queries and outputs. </span><span class="c12">function_blast.sh</span><span>&nbsp;creates output folders (</span><span class="c9">data/blast/Bumble_bees_proteins, data/blast/Honey_bees_proteins, data/blast/Bumble_Honey_bees_proteins</span><span>) and assigns variables used in </span><span class="c12">blast_gene_families.sh</span><span>&nbsp;(variables correspond to every </span><span class="c6">Gene_family_*.fasta</span><span>&nbsp;files present in each folder of group bees and stored in </span><span class="c6">list_files.txt</span><span class="c0">).</span></p><p class="c8"><span class="c6 c2">Inputs: outputs of </span><span class="c12 c2">proteinseqextract.py</span><span class="c2">&nbsp;- </span><span class="c6 c2">Gene_family_*.fasta</span><span class="c2">&nbsp;in </span><span class="c2 c9">data/Bumble_bees_proteins, data/Honey_bees_proteins, data/Bumble_Honey_bees_proteins</span><span class="c2">&nbsp;(to select </span><span class="c6 c2">Gene_family_*.fasta</span><span class="c2">, </span><span class="c12 c2">function_blast.sh</span><span class="c2">&nbsp;will go through </span><span class="c6 c2">list_files.txt</span><span class="c2">&nbsp;- created in the terminal: </span><span class="c4 c2">ls Gene_family_* &gt; list_files.txt</span><span class="c5 c2">)</span></p><p class="c8"><span class="c6 c2">Outputs: Gene_family_*.out </span><span class="c2">of its corresponding </span><span class="c6 c2">Gene_family_*.fasta</span><span class="c2">&nbsp;of its corresponding folder of group of bees </span><span class="c2 c9">(data/Bumble_bees_proteins, data/Honey_bees_proteins, data/Bumble_Honey_bees_proteins)</span><span class="c2">&nbsp;in its corresponding output folders </span><span class="c2 c9">(data/blast/Bumble_bees_proteins, data/blast/Honey_bees_proteins, data/blast/Bumble_Honey_bees_proteins)</span><span class="c2">&nbsp;- each </span><span class="c6 c2">Gene_family_*.out</span><span class="c2">&nbsp;file contains blast hits for each reference_genome|protein_ID&rsquo;s sequence - for each reference_genome|protein_ID&rsquo;s sequence (query): 5 first lines summarizing blastp information (version of BLASTP, query, database, fields and the total number of found hits) - fields correspond to the name of each resulting column (total of 14 columns) - subsequent lines correspond to every hits of the query</span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span>4. For every blast results of each reference_genome|protein_ID&rsquo;s sequence (query) present in each </span><span class="c6">Gene_family_*.out</span><span>&nbsp;files in each folder of group of bees (</span><span class="c9">data/blast/Bumble_bees_proteins, data/blast/Honey_bees_proteins, data/blast/Bumble_Honey_bees_proteins</span><span>), </span><span class="c12">blast_hits_extract.py</span><span class="c0">&nbsp;parses hits according to different parameters (threshold_alignment_length, threshold_ID, threshold_eval, threshold_bitscore): </span></p><ul class="c7 lst-kix_lnlbj3dhp3t-0 start"><li class="c8 c11"><span class="c0">threshold_alignment_length : constant = 0.8</span></li><li class="c8 c11"><span class="c0">threshold_ID : starts at 50</span></li><li class="c8 c11"><span class="c0">threshold _eval : 0.00001</span></li><li class="c8 c11"><span class="c0">threshold_bitscore : 0</span></li></ul><p class="c1"><span class="c0"></span></p><p class="c8"><span>For a given </span><span class="c6">Gene_family_*.out</span><span>, if there are less than 25 blast hits with parameters below, we relax the threshold_ID by multiplying it by 0.8 until reaching at least 25 blast hits. </span><span class="c12">blast_hits_extract.sh</span><span>&nbsp;creates output folders (</span><span class="c9">data/parsed_blast/Bumble_bees_proteins, data/parsed_blast/Honey_bees_proteins, data/parsed_blast/Bumble_Honey_bees_proteins</span><span class="c0">) and assigns parameters as below. </span></p><p class="c8"><span class="c6 c2">Inputs: outputs of </span><span class="c12 c2">blast_gene_families.sh</span><span class="c2">&nbsp;and </span><span class="c12 c2">function_blast.sh</span><span class="c2">&nbsp;- </span><span class="c6 c2">Gene_family_*.out</span><span class="c2">&nbsp;in </span><span class="c2 c9">data/blast/Bumble_bees_proteins, data/blast/Honey_bees_proteins, data/blast/Bumble_Honey_bees_proteins</span><span class="c2">&nbsp;(to select </span><span class="c6 c2">Gene_family_*.out</span><span class="c2">, </span><span class="c2 c12">blast_hits_extract.py</span><span class="c2">&nbsp;will go through </span><span class="c6 c2">list_files.txt </span><span class="c2">- created in the terminal: </span><span class="c4 c2">ls Gene_family_* &gt; list_files.txt</span><span class="c5 c2">)</span></p><p class="c8"><span class="c6 c2">Outputs: Gene_family_*_parsed.out </span><span class="c2">of its corresponding </span><span class="c6 c2">Gene_family_*.out</span><span class="c2">&nbsp;of its corresponding folder of group of bees </span><span class="c2 c9">(data/blast/Bumble_bees_proteins, data/blast/Honey_bees_proteins, data/blast/Bumble_Honey_bees_proteins)</span><span class="c2">&nbsp;in its corresponding output folders </span><span class="c2 c9">(data/parsed_blast/Bumble_bees_proteins, data/parsed_blast/Honey_bees_proteins, data/parsed_blast/Bumble_Honey_bees_proteins)</span><span class="c2">&nbsp;- each </span><span class="c6 c2">Gene_family_*_parsed.out</span><span class="c5 c2">&nbsp;file contains the &lsquo;best&rsquo; blast hits for each reference_genome|protein_ID&rsquo;s sequence - the header is &lsquo;# Query_ID&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Subject_titles&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;%_Identity&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Alignment_length&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;evalue&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;bit_score&rsquo; - for each &lsquo;best&rsquo; hits, informations are stored in its corresponding column</span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span>5. From every </span><span class="c4">parsed</span><span>&nbsp;blast results of each reference_genome|protein_ID&rsquo;s sequence (query) present in each </span><span class="c6">Gene_family_*_parsed.out</span><span>&nbsp;files in each folder of group of bees (</span><span class="c9">data/parsed_blast/Bumble_bees_proteins, data/parsed_blast/Honey_bees_proteins, data/parsed_blast/Bumble_Honey_bees_proteins</span><span>), </span><span class="c12">extract_taxonomy_hierarchy.py</span><span>&nbsp;first creates a list of subject IDs (hit IDs, for example &lsquo;WP_052726720&rsquo;). Subsequently, thanks to Entrez (from Bio package), the function extracts the hierarchical taxonomy of each subject IDs (from protein database of NCBI, for example: &#39;Bacteria;Firmicutes;Bacilli;Lactobacillales;Lactobacillaceae;Lactobacillus&#39;). </span><span class="c2">The example (&lsquo;WP_052726720&rsquo;) is used as reference hierarchical taxonomy.</span><span class="c0">&nbsp;Finally, the function looks for the most recent taxa which is shared between each hierarchical taxonomy of each subject IDs and the reference hierarchical taxonomy. Hierarchical taxonomy distance are subjective:</span></p><ul class="c7 lst-kix_40tprdp2abg-0 start"><li class="c8 c11"><span class="c0">Lactobacillus = 1</span></li><li class="c8 c11"><span class="c0">Lactobacillaceae = 2</span></li><li class="c8 c11"><span class="c0">Lactobacillales = 3</span></li><li class="c8 c11"><span class="c0">Bacilli = 4</span></li><li class="c8 c11"><span class="c0">Firmicutes = 5</span></li><li class="c8 c11"><span class="c0">Bacteria = 6</span></li><li class="c8 c11"><span class="c0">None = 7 (which either corresponds to Archae or Eukaryota - contaminations)</span></li></ul><p class="c1"><span class="c0"></span></p><p class="c8"><span>The function creates a summary file (</span><span class="c6">hierarchical_taxonomy.txt</span><span>) which is located in </span><span class="c9">data/parsed_blast</span><span class="c0">.</span></p><p class="c8"><span class="c6 c2">Inputs: Gene_family_*_parsed.out </span><span class="c2">of its corresponding folder of group of bees </span><span class="c2 c9">(data/parsed_blast/Bumble_bees_proteins, data/parsed_blast/Honey_bees_proteins, data/parsed_blast/Bumble_Honey_bees_proteins)</span><span class="c5 c2">&nbsp;</span></p><p class="c8"><span class="c6 c2">Outputs: hierarchical_taxonomy.txt</span><span class="c2">&nbsp;in </span><span class="c2 c9">data/parsed_blast </span><span class="c5 c2">- the first column corresponds to the strain IDs, the second one corresponds to the hierarchical taxonomy and the third one corresponds to the subjective hierarchical taxonomy distance</span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span>6. For every </span><span class="c4">parsed</span><span>&nbsp;blast results of each reference_genome|protein_ID&rsquo;s sequence (query) present in each </span><span class="c6">Gene_family_*_parsed.out</span><span>&nbsp;files in each folder of group of bees (</span><span class="c9">data/parsed_blast/Bumble_bees_proteins, data/parsed_blast/Honey_bees_proteins, data/parsed_blast/Bumble_Honey_bees_proteins</span><span>), </span><span class="c12">plot_identity_taxonomy.R</span><span>&nbsp;extracts its percentage of identity and its hierarchical taxonomy distance (in </span><span class="c6">hierarchical_taxonomy.txt</span><span>) and plot them against each other. Moreover, for each </span><span class="c6">Gene_family_*_parsed.out</span><span class="c0">, on one hand, we will perform linear regression model in addition to polynomial model and investigate if models differ. If they differ, it would mean that polynomial model is the best one and suggest potential horizontal gene transfer. </span></p><p class="c8"><span class="c6 c2">Inputs: Gene_family_*_parsed.out </span><span class="c2">of its corresponding folder of group of bees </span><span class="c2 c9">(data/parsed_blast/Bumble_bees_proteins, data/parsed_blast/Honey_bees_proteins, data/parsed_blast/Bumble_Honey_bees_proteins)</span><span class="c2">&nbsp;and </span><span class="c6 c2">hierarchical_taxonomy.txt</span><span class="c2">&nbsp;in </span><span class="c2 c9">data/parsed_blast</span></p><p class="c8"><span class="c6 c2">Outputs: Bumble_taxo_plot.pdf, Honey_taxo_plot.pdf, Bumble_Honey_taxo_plot.pdf</span><span class="c2">&nbsp;and </span><span class="c6 c2">potential_HGT.txt</span><span class="c2">&nbsp;in </span><span class="c2 c9">data/parsed_blast</span><span class="c2">&nbsp;- each </span><span class="c6 c2">*.pdf</span><span class="c2">&nbsp;file contains all plots of a given bee group - </span><span class="c6 c2">potential_HGT.txt</span><span class="c2">&nbsp;contains a list of </span><span class="c6 c2">Gene_family_*_parsed.out</span><span class="c5 c2">&nbsp;that are orthologous genes potentially acquired by HGT</span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span>7. Manually, we decide which gene families are good candidates to investigate if orthologous genes are acquired by HGT. These gene family candidates are listed in </span><span class="c6">list_files_HGT_to_trees.txt</span><span>&nbsp;(in </span><span class="c9">data/parsed_blast</span><span>). </span><span class="c6">Gene_family_1058_parsed.out</span><span>&nbsp;and </span><span class="c6">Gene_family_991_parsed.out</span><span>&nbsp;(</span><span class="c9">data/parsed_blast/Bumble_Honey_bees_proteins</span><span>) correspond to &lsquo;confirmed&rsquo; HGT candidates. </span><span class="c6">Gene_family_1099_parsed.out</span><span>&nbsp;(</span><span class="c9">data/parsed_blast/Bumble_Honey_bees_proteins</span><span>) and </span><span class="c6">Gene_family_1674_parsed.out </span><span>(</span><span class="c9">data/parsed_blast/Bumble_bees_proteins</span><span>) correspond to &lsquo;supposed&rsquo; HGT candidates. </span><span class="c6">Gene_family_1048_parsed.out</span><span>&nbsp;(</span><span class="c9">data/parsed_blast/Bumble_Honey_bees_proteins</span><span>) corresponds to non-HGT candidate. First, </span><span class="c12">extract_seq_blast_hits.py</span><span>&nbsp;extracts the protein sequence of every blast hits of selected </span><span class="c6">Gene_family_*_parsed.out</span><span>&nbsp;(listed in </span><span class="c6">list_files.txt</span><span>). From this, </span><span class="c12">align_potential_HGT.sh</span><span>&nbsp;aligns the sequence of every blast hits of selected </span><span class="c6">Gene_family_*_parsed.out</span><span>&nbsp;(listed in </span><span class="c6">list_files.txt</span><span>) in its folder of group of bees (</span><span class="c9">data/phylogeny_potential_HGT/Bumble_bees_proteins, data/phylogeny_potential_HGT/Honey_bees_proteins, data/phylogeny_potential_HGT/Bumble_Honey_bees_proteins</span><span>). Then, </span><span class="c12">tree_wo_bootstrap.sh</span><span>&nbsp;runs RAXML tool to infer phylogenetic trees (without bootstrapping) of all aligned protein sequences of every blast hits of selected </span><span class="c6">Gene_family_*_parsed.out</span><span>&nbsp; (in its bee folder in </span><span class="c9">data/phylogeny_potential_HGT</span><span class="c0">). </span></p><p class="c8"><span class="c12">extract_seq_blast_hits.py</span><span>:</span></p><p class="c8"><span class="c6 c2">Inputs: list_files.txt</span><span class="c2">&nbsp;in </span><span class="c2 c9">data/parsed_blast</span><span class="c2">, </span><span class="c6 c2">Gene_family_*_parsed.out </span><span class="c2">of its corresponding folder of group of bees </span><span class="c2 c9">(data/parsed_blast/Bumble_bees_proteins, data/parsed_blast/Honey_bees_proteins, data/parsed_blast/Bumble_Honey_bees_proteins) </span><span class="c2">listed in </span><span class="c6 c2">list_files.txt</span></p><p class="c8"><span class="c6 c2">Outputs: amino_acid_seq_Gene_family_*_parsed.fasta </span><span class="c2">in </span><span class="c2 c9">data/parsed_blast</span><span class="c2">&nbsp;- each output file contains protein sequences of every blast hits of a selected gene family. The ID name of each sequence is defined as follows: StrainName_SujectID_HierarchicalTaxonomyDistance</span></p><p class="c8"><span class="c12">align_potential_HGT.sh</span><span>:</span></p><p class="c8"><span class="c6 c2">Inputs: amino_acid_seq_Gene_family_*_parsed.fasta </span><span class="c2">in </span><span class="c2 c9">data/parsed_blast</span></p><p class="c8"><span class="c6 c2">Outputs: Gene_family_*_parsed.multifasta</span><span class="c2">&nbsp;in its corresponding folder of group of bees </span><span class="c2 c9">(data/phylogeny_potential_HGT/Bumble_bees_proteins, data/phylogeny_potential_HGT/Honey_bees_proteins, data/phylogeny_potential_HGT/Bumble_Honey_bees_proteins)</span><span class="c5 c2">&nbsp;- each output file contains aligned protein sequences of every blast hits of a selected gene family</span></p><p class="c8"><span class="c12">tree_wo_bootstrap.sh</span><span>:</span></p><p class="c8"><span class="c6 c2">Inputs: Gene_family_*_parsed.multifasta</span><span class="c2">&nbsp;in its corresponding folder of group of bees </span><span class="c18 c2 c9">(data/phylogeny_potential_HGT/Bumble_bees_proteins, data/phylogeny_potential_HGT/Honey_bees_proteins, data/phylogeny_potential_HGT/Bumble_Honey_bees_proteins)</span></p><p class="c8"><span class="c6 c2">Outputs: *ML_wo_bootstrap_Gene_family_*_parsed.out </span><span class="c2">in </span><span class="c2 c9 c18">data/phylogeny_potential_HGT/RAxML_results</span></p><p class="c8"><span class="c0">&gt;&gt; RAxML_bestTree.ML_wo_bootstrap_Gene_family_*_parsed.out can then be opened using FigTree tool (available at http://tree.bio.ed.ac.uk/software/figtree/)</span></p><p class="c1"><span class="c0"></span></p><p class="c8"><span>8. The inference of genomic islands is performed using Islandviewer tool (available at http://www.pathogenomics.sfu.ca/islandviewer/). For this purpose, two reference genomes from Bumble bees group (</span><span class="c4 c2">F5_237</span><span>&nbsp;and </span><span class="c2 c4">F5_245</span><span>) and one from Honey bees group (</span><span class="c4 c2">Lb_183</span><span>) are analysed. For each referenge genome, </span><span class="c12">extract_coordinates_function_from_gbkfiles.py</span><span>&nbsp;extracts the protein name of every genes present in each Gene_family_* from </span><span class="c6">GeneFamilies.txt </span><span>(in </span><span class="c16">data</span><span>). Thanks to GenBank files (</span><span class="c16">mls_2016/genome_files</span><span>), the coordinates and the function of each protein are recovered. Then, we map manually the specific position of the protein onto its reference genome. Results are present in </span><span class="c9">data/IslandViewerResults, </span><span class="c6">*.png</span><span class="c3">&nbsp;</span><span>shows the circular representation of each reference genome and the position of the genomic islands, &nbsp;</span><span class="c6">*.tsv</span><span class="c3">&nbsp;</span><span>contains the coordinates of each genomic island and which method allows this inference. Moreover, </span><span class="c6">MappingIntoGenome.txt</span><span class="c3">&nbsp;</span><span class="c0">sums up the different results for this part.</span></p><p class="c8"><span class="c6 c2">Input: GeneFamilies.txt</span><span class="c2">&nbsp;in </span><span class="c2 c9">data</span><span class="c2">, </span><span class="c6 c2">GenBank files</span><span class="c2">&nbsp;in </span><span class="c16 c2">mls_2016/genome_files</span></p><p class="c8"><span class="c6 c2">Outputs: *.png files</span><span class="c2">&nbsp;, </span><span class="c2 c6">*.tsv files </span><span class="c2">and</span><span class="c6 c2">&nbsp;MappingIntoGenome.txt </span><span class="c2">in</span><span class="c6 c2">&nbsp;</span><span class="c2 c9">data/IslandViewerResults</span></p><p class="c1"><span class="c0"></span></p></body></html>