<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns:mv="http://macVmlSchemaUri" xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta http-equiv=Content-Type content="text/html; charset=macintosh">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 14">
<meta name=Originator content="Microsoft Word 14">
<link rel=File-List href="Documentation_SAGE_JC_VR_files/filelist.xml">
<link rel=themeData href="Documentation_SAGE_JC_VR_files/themedata.xml">
<!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:Zoom>140</w:Zoom>
  <w:TrackMoves/>
  <w:TrackFormatting/>
  <w:PunctuationKerning/>
  <w:ValidateAgainstSchemas/>
  <w:SaveIfXMLInvalid>false</w:SaveIfXMLInvalid>
  <w:IgnoreMixedContent>false</w:IgnoreMixedContent>
  <w:AlwaysShowPlaceholderText>false</w:AlwaysShowPlaceholderText>
  <w:DoNotPromoteQF/>
  <w:LidThemeOther>FR-CH</w:LidThemeOther>
  <w:LidThemeAsian>JA</w:LidThemeAsian>
  <w:LidThemeComplexScript>X-NONE</w:LidThemeComplexScript>
  <w:Compatibility>
   <w:BreakWrappedTables/>
   <w:SnapToGridInCell/>
   <w:WrapTextWithPunct/>
   <w:UseAsianBreakRules/>
   <w:DontGrowAutofit/>
   <w:SplitPgBreakAndParaMark/>
  </w:Compatibility>
  <m:mathPr>
   <m:mathFont m:val="Cambria Math"/>
   <m:brkBin m:val="before"/>
   <m:brkBinSub m:val="&#45;-"/>
   <m:smallFrac m:val="off"/>
   <m:dispDef/>
   <m:lMargin m:val="0"/>
   <m:rMargin m:val="0"/>
   <m:defJc m:val="centerGroup"/>
   <m:wrapIndent m:val="1440"/>
   <m:intLim m:val="subSup"/>
   <m:naryLim m:val="undOvr"/>
  </m:mathPr></w:WordDocument>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:LatentStyles DefLockedState="false" DefUnhideWhenUsed="true"
  DefSemiHidden="true" DefQFormat="false" DefPriority="99"
  LatentStyleCount="276">
  <w:LsdException Locked="false" Priority="0" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Normal"/>
  <w:LsdException Locked="false" Priority="9" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="heading 1"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 2"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 3"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 4"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 5"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 6"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 7"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 8"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 9"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 1"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 2"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 3"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 4"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 5"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 6"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 7"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 8"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 9"/>
  <w:LsdException Locked="false" Priority="35" QFormat="true" Name="caption"/>
  <w:LsdException Locked="false" Priority="10" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Title"/>
  <w:LsdException Locked="false" Priority="1" Name="Default Paragraph Font"/>
  <w:LsdException Locked="false" Priority="11" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Subtitle"/>
  <w:LsdException Locked="false" Priority="22" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Strong"/>
  <w:LsdException Locked="false" Priority="20" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Emphasis"/>
  <w:LsdException Locked="false" Priority="59" SemiHidden="false"
   UnhideWhenUsed="false" Name="Table Grid"/>
  <w:LsdException Locked="false" UnhideWhenUsed="false" Name="Placeholder Text"/>
  <w:LsdException Locked="false" Priority="1" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="No Spacing"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 1"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 1"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 1"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 1"/>
  <w:LsdException Locked="false" UnhideWhenUsed="false" Name="Revision"/>
  <w:LsdException Locked="false" Priority="34" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="List Paragraph"/>
  <w:LsdException Locked="false" Priority="29" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Quote"/>
  <w:LsdException Locked="false" Priority="30" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Intense Quote"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 1"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 1"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 1"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 1"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 1"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 2"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 2"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 2"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 2"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 2"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 2"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 2"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 2"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 3"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 3"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 3"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 3"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 3"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 3"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 3"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 3"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 4"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 4"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 4"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 4"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 4"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 4"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 4"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 4"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 5"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 5"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 5"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 5"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 5"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 5"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 5"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 5"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 6"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 6"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 6"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 6"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 6"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 6"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 6"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 6"/>
  <w:LsdException Locked="false" Priority="19" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Subtle Emphasis"/>
  <w:LsdException Locked="false" Priority="21" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Intense Emphasis"/>
  <w:LsdException Locked="false" Priority="31" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Subtle Reference"/>
  <w:LsdException Locked="false" Priority="32" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Intense Reference"/>
  <w:LsdException Locked="false" Priority="33" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Book Title"/>
  <w:LsdException Locked="false" Priority="37" Name="Bibliography"/>
  <w:LsdException Locked="false" Priority="39" QFormat="true" Name="TOC Heading"/>
 </w:LatentStyles>
</xml><![endif]-->
<style>
<!--
 /* Font Definitions */
@font-face
	{font-family:Arial;
	panose-1:2 11 6 4 2 2 2 2 2 4;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:-536859905 -1073711037 9 0 511 0;}
@font-face
	{font-family:Arial;
	panose-1:2 11 6 4 2 2 2 2 2 4;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:-536859905 -1073711037 9 0 511 0;}
 /* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	margin:0cm;
	margin-bottom:.0001pt;
	line-height:115%;
	mso-pagination:widow-orphan;
	font-size:11.0pt;
	font-family:Arial;
	mso-fareast-font-family:Arial;
	color:black;}
h1
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	margin-top:20.0pt;
	margin-right:0cm;
	margin-bottom:6.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:1;
	font-size:20.0pt;
	font-family:Arial;
	color:black;
	mso-font-kerning:0pt;
	font-weight:normal;}
h1.CxSpFirst
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:20.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:1;
	font-size:20.0pt;
	font-family:Arial;
	color:black;
	mso-font-kerning:0pt;
	font-weight:normal;}
h1.CxSpMiddle
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:1;
	font-size:20.0pt;
	font-family:Arial;
	color:black;
	mso-font-kerning:0pt;
	font-weight:normal;}
h1.CxSpLast
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:6.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:1;
	font-size:20.0pt;
	font-family:Arial;
	color:black;
	mso-font-kerning:0pt;
	font-weight:normal;}
h2
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	margin-top:18.0pt;
	margin-right:0cm;
	margin-bottom:6.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:2;
	font-size:16.0pt;
	font-family:Arial;
	color:black;
	font-weight:normal;}
h2.CxSpFirst
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:18.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:2;
	font-size:16.0pt;
	font-family:Arial;
	color:black;
	font-weight:normal;}
h2.CxSpMiddle
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:2;
	font-size:16.0pt;
	font-family:Arial;
	color:black;
	font-weight:normal;}
h2.CxSpLast
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:6.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:2;
	font-size:16.0pt;
	font-family:Arial;
	color:black;
	font-weight:normal;}
h3
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	margin-top:16.0pt;
	margin-right:0cm;
	margin-bottom:4.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:3;
	font-size:14.0pt;
	font-family:Arial;
	color:#434343;
	font-weight:normal;}
h3.CxSpFirst
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:16.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:3;
	font-size:14.0pt;
	font-family:Arial;
	color:#434343;
	font-weight:normal;}
h3.CxSpMiddle
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:3;
	font-size:14.0pt;
	font-family:Arial;
	color:#434343;
	font-weight:normal;}
h3.CxSpLast
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:4.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:3;
	font-size:14.0pt;
	font-family:Arial;
	color:#434343;
	font-weight:normal;}
h4
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	margin-top:14.0pt;
	margin-right:0cm;
	margin-bottom:4.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:4;
	font-size:12.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;}
h4.CxSpFirst
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:14.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:4;
	font-size:12.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;}
h4.CxSpMiddle
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:4;
	font-size:12.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;}
h4.CxSpLast
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:4.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:4;
	font-size:12.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;}
h5
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	margin-top:12.0pt;
	margin-right:0cm;
	margin-bottom:4.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:5;
	font-size:11.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;}
h5.CxSpFirst
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:12.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:5;
	font-size:11.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;}
h5.CxSpMiddle
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:5;
	font-size:11.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;}
h5.CxSpLast
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:4.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:5;
	font-size:11.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;}
h6
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	margin-top:12.0pt;
	margin-right:0cm;
	margin-bottom:4.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:6;
	font-size:11.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;
	font-style:italic;
	mso-bidi-font-style:normal;}
h6.CxSpFirst
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:12.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:6;
	font-size:11.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;
	font-style:italic;
	mso-bidi-font-style:normal;}
h6.CxSpMiddle
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:6;
	font-size:11.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;
	font-style:italic;
	mso-bidi-font-style:normal;}
h6.CxSpLast
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:4.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:6;
	font-size:11.0pt;
	font-family:Arial;
	color:#666666;
	font-weight:normal;
	font-style:italic;
	mso-bidi-font-style:normal;}
p.MsoTitle, li.MsoTitle, div.MsoTitle
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:3.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:26.0pt;
	font-family:Arial;
	mso-fareast-font-family:Arial;
	color:black;}
p.MsoTitleCxSpFirst, li.MsoTitleCxSpFirst, div.MsoTitleCxSpFirst
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:26.0pt;
	font-family:Arial;
	mso-fareast-font-family:Arial;
	color:black;}
p.MsoTitleCxSpMiddle, li.MsoTitleCxSpMiddle, div.MsoTitleCxSpMiddle
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:26.0pt;
	font-family:Arial;
	mso-fareast-font-family:Arial;
	color:black;}
p.MsoTitleCxSpLast, li.MsoTitleCxSpLast, div.MsoTitleCxSpLast
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:3.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:26.0pt;
	font-family:Arial;
	mso-fareast-font-family:Arial;
	color:black;}
p.MsoSubtitle, li.MsoSubtitle, div.MsoSubtitle
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:16.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:15.0pt;
	font-family:Arial;
	mso-fareast-font-family:Arial;
	color:#666666;}
p.MsoSubtitleCxSpFirst, li.MsoSubtitleCxSpFirst, div.MsoSubtitleCxSpFirst
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:15.0pt;
	font-family:Arial;
	mso-fareast-font-family:Arial;
	color:#666666;}
p.MsoSubtitleCxSpMiddle, li.MsoSubtitleCxSpMiddle, div.MsoSubtitleCxSpMiddle
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:15.0pt;
	font-family:Arial;
	mso-fareast-font-family:Arial;
	color:#666666;}
p.MsoSubtitleCxSpLast, li.MsoSubtitleCxSpLast, div.MsoSubtitleCxSpLast
	{mso-style-unhide:no;
	mso-style-parent:normal;
	mso-style-next:normal;
	mso-style-type:export-only;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:16.0pt;
	margin-left:0cm;
	mso-add-space:auto;
	line-height:115%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:15.0pt;
	font-family:Arial;
	mso-fareast-font-family:Arial;
	color:#666666;}
a:link, span.MsoHyperlink
	{mso-style-noshow:yes;
	mso-style-priority:99;
	color:blue;
	mso-themecolor:hyperlink;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{mso-style-noshow:yes;
	mso-style-priority:99;
	color:purple;
	mso-themecolor:followedhyperlink;
	text-decoration:underline;
	text-underline:single;}
p.normal, li.normal, div.normal
	{mso-style-name:normal;
	mso-style-unhide:no;
	mso-style-parent:"";
	margin:0cm;
	margin-bottom:.0001pt;
	line-height:115%;
	mso-pagination:widow-orphan;
	font-size:11.0pt;
	font-family:Arial;
	mso-fareast-font-family:Arial;
	color:black;}
.MsoChpDefault
	{mso-style-type:export-only;
	mso-default-props:yes;
	font-size:11.0pt;
	mso-ansi-font-size:11.0pt;
	mso-bidi-font-size:11.0pt;
	font-family:Arial;
	mso-ascii-font-family:Arial;
	mso-fareast-font-family:Arial;
	mso-hansi-font-family:Arial;
	mso-bidi-font-family:Arial;
	color:black;}
.MsoPapDefault
	{mso-style-type:export-only;
	line-height:115%;}
@page WordSection1
	{size:595.45pt 841.7pt;
	margin:72.0pt 72.0pt 72.0pt 72.0pt;
	mso-header-margin:0cm;
	mso-footer-margin:36.0pt;
	mso-page-numbers:1;
	mso-paper-source:0;}
div.WordSection1
	{page:WordSection1;}
 /* List Definitions */
@list l0
	{mso-list-id:581178886;
	mso-list-template-ids:514499900;}
@list l0:level1
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:18.0pt;
	text-decoration:none;
	text-underline:none;}
@list l0:level2
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:54.0pt;
	text-decoration:none;
	text-underline:none;}
@list l0:level3
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:90.0pt;
	text-decoration:none;
	text-underline:none;}
@list l0:level4
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:126.0pt;
	text-decoration:none;
	text-underline:none;}
@list l0:level5
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:162.0pt;
	text-decoration:none;
	text-underline:none;}
@list l0:level6
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:198.0pt;
	text-decoration:none;
	text-underline:none;}
@list l0:level7
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:234.0pt;
	text-decoration:none;
	text-underline:none;}
@list l0:level8
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:270.0pt;
	text-decoration:none;
	text-underline:none;}
@list l0:level9
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:306.0pt;
	text-decoration:none;
	text-underline:none;}
@list l1
	{mso-list-id:626467189;
	mso-list-template-ids:-2077334298;}
@list l1:level1
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:18.0pt;
	text-decoration:none;
	text-underline:none;}
@list l1:level2
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:54.0pt;
	text-decoration:none;
	text-underline:none;}
@list l1:level3
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:90.0pt;
	text-decoration:none;
	text-underline:none;}
@list l1:level4
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:126.0pt;
	text-decoration:none;
	text-underline:none;}
@list l1:level5
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:162.0pt;
	text-decoration:none;
	text-underline:none;}
@list l1:level6
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:198.0pt;
	text-decoration:none;
	text-underline:none;}
@list l1:level7
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:234.0pt;
	text-decoration:none;
	text-underline:none;}
@list l1:level8
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:270.0pt;
	text-decoration:none;
	text-underline:none;}
@list l1:level9
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:306.0pt;
	text-decoration:none;
	text-underline:none;}
ol
	{margin-bottom:0cm;}
ul
	{margin-bottom:0cm;}
-->
</style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
table.MsoNormalTable
	{mso-style-name:"Table Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-priority:99;
	mso-style-parent:"";
	mso-padding-alt:0cm 5.4pt 0cm 5.4pt;
	mso-para-margin:0cm;
	mso-para-margin-bottom:.0001pt;
	line-height:115%;
	mso-pagination:widow-orphan;
	font-size:11.0pt;
	font-family:Arial;
	color:black;}
</style>
<![endif]--><!--[if gte mso 9]><xml>
 <o:shapedefaults v:ext="edit" spidmax="1026"/>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <o:shapelayout v:ext="edit">
  <o:idmap v:ext="edit" data="1"/>
 </o:shapelayout></xml><![endif]-->
</head>

<body lang=FR-CH link=blue vlink=purple style='tab-interval:36.0pt'>

<div class=WordSection1>

<h1 style='text-align:justify'><a name="_fr11brr5f70d"></a>Report Group 5</h1>

<p class=normal align=center style='text-align:center'><b style='mso-bidi-font-weight:
normal'><u><span style='font-size:16.0pt;line-height:115%'>Ancient HGT to Firm5
(from unknown donor)<o:p></o:p></span></u></b></p>

<p class=normal align=center style='text-align:center'><b style='mso-bidi-font-weight:
normal'><span style='font-size:16.0pt;line-height:115%'>Joaquim, Virginie &amp;
Kamil<o:p></o:p></span></b></p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'>We are going to investigate which
genes of our Firm5 group are potentially acquired by ancient horizontal gene
transfer from unknown donor. </p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'>Our GitHub repository is: <a
href="https://github.com/KamilSJaron/SAGE_Firm5_specific_HGT.git"><span
style='font-size:12.0pt;line-height:115%;color:#0366D6;text-decoration:none;
text-underline:none'>https://github.com/KamilSJaron/SAGE_Firm5_specific_HGT.git</span></a></p>

<p class=normal style='text-align:justify'>On Vital-IT, you can access the same
repo by typing: </p>

<p class=normal style='text-align:justify;text-indent:36.0pt'><i
style='mso-bidi-font-style:normal'>cd
/scratch/beegfs/monthly/mls_2016/claivaz_ricci/SAGE_Firm5_specific_HGT<o:p></o:p></i></p>

<p class=normal style='text-align:justify'>In this folder, README.md comprises
all informations about our <span style='color:#980000'>folders</span> and <span
style='color:#38761D'>files</span>. To make it short, <span style='color:#980000'>data</span>
contains all inputs and outputs files whereas <span style='color:#980000'>scripts</span>
contains all <span style='color:blue'>scripts</span> for data analysis. </p>

<p class=normal style='text-align:justify'><span style='color:#980000'>Folders
are in red</span>, <span style='color:#38761D'>files (inputs and outputs) in
green</span> and <span style='color:blue'>scripts in blue</span>.</p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'><b style='mso-bidi-font-weight:normal'><u><span
style='font-size:14.0pt;line-height:115%'>Summary:<o:p></o:p></span></u></b></p>

<p class=normal style='text-align:justify'>We are going to select all gene
families which contains orthologous genes if at least 80% of all strains in a
given group of bees are present. Then, we are going to extract each protein
sequences of all orthologous genes of each group of bees and blast them against
RefSeq database (ncbi database). Then, we are going to parse blast results
according to parameters. Thanks to blast results, we are going to determine
hierarchical taxonomy and hierarchical taxonomy distance for each blast hit.
This will allow us to plot % of identity of blast hits against hierarchical
taxonomy distance. In the end, all bast hits corresponding to ‘Lactobacillus’
(in addition to supposed contaminations) will be discarded as we are interested
in HGT from species other than Lactobacillus. From blast results, we will
perform 2 different analysis. The first one is constructing phylogenetic trees
of gene families potentially acquired by HGT and the second one is to predict
genomic islands of gene families potentially acquired by HGT.</p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'><b style='mso-bidi-font-weight:normal'><u><span
style='font-size:14.0pt;line-height:115%'>Group of bees:<o:p></o:p></span></u></b></p>

<p class=normal style='text-align:justify'><b style='mso-bidi-font-weight:normal'>Bumble_bees:</b>
B_pascuorum+B_bohemicus </p>

<p class=normal style='text-align:justify'>('F225','F230','F233','F234','F236','F237')+('F228',
'F245','F246','F247’)</p>

<p class=normal style='text-align:justify'><b style='mso-bidi-font-weight:normal'>Honey_bees:</b>
A_mellifera </p>

<p class=normal style='text-align:justify'>('JF72','JF73','JG30','JF74','JF75','JF76','F259','F260','F261','F262','F263','L183','L184','L185','L186')</p>

<p class=normal style='text-align:justify'><b style='mso-bidi-font-weight:normal'>Bumble_Honey_bees:</b>
Bumble_bees+Honey_bees</p>

<p class=normal style='text-align:justify'><b style='mso-bidi-font-weight:normal'>Outgroup:</b>
other strains which are not Bumble_bees and Honey_bees<span
style="mso-spacerun:yes">&nbsp; </span></p>

<p class=normal style='text-align:justify'>('JG29','LA14','LA2','LDB','LGAS','LHV','LJP','WANG')</p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'><b style='mso-bidi-font-weight:normal'><u><span
style='font-size:14.0pt;line-height:115%'>Steps:</span></u></b></p>

<p class=normal style='text-align:justify'>1. For each group of bees
(Bumble_bees, Honey_bees, Bumble_Honey_bees, Outgroup), <span style='color:
blue'>subgroup_sort.py</span> looks into each Gene_family_* (each line) of <span
style='color:#38761D'>GeneFamilies.txt</span>. If a Gene_family_* line contains
at least 80% of all strains present in the group of bees, it stores the
Gene_family_* line into its corresponding file. </p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Input: GeneFamilies.txt </span><span style='color:#980000'>(in
data)<o:p></o:p></span></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Outputs: Bumble_bees.txt, Honey_bees.txt,
Bumble_Honey_bees.txt and Outgroup.txt </span><span style='color:#980000'>(in
data/sort_group)<o:p></o:p></span></i></p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'>2. For each group of bees and for
each Gene_family_*, <span style='color:blue'>proteinseqextract.py</span>
extracts all protein sequences for further ‘Blast’ step. This function creates
a folder for each group of bees which comprises fasta files for each
Gene_family_* containing protein sequences. </p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Inputs: outputs of </span><span style='color:blue'>subgroup_sort.py</span><span
style='color:#38761D'> </span>in<span style='color:#980000'> data/sort_group</span><span
style='color:#38761D'> (Bumble_bees.txt, Honey_bees.txt, Bumble_Honey_bees.txt
or Outgroup.txt), all_proteins.fasta</span> <span style='color:#980000'>(in
mls_2016/blast folder)</span><span style='color:#38761D'> <o:p></o:p></span></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Outputs: </span><span style='color:#980000'>folders in
data of Bumble_bees_proteins, Honey_bees_proteins, Bumble_Honey_bees_proteins</span>
- each folder contains .fasta files of gene families <span style='color:#38761D'>(Gene_family_*.fasta)</span>
- a file concerns one gene family and contains all reference_genome|protein_ID
and their sequence in fasta format<o:p></o:p></i></p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'>3. For every protein sequences
present in each <span style='color:#38761D'>Gene_family_*.fasta</span> files in
each folder of group of bees (<span style='color:#980000'>data/Bumble_bees_proteins,
data/Honey_bees_proteins, data/Bumble_Honey_bees_proteins</span>), <span
style='color:blue'>blast_gene_families.sh</span> and <span style='color:blue'>function_blast.sh</span>
perform blastp against RefSeq database (module Blast is previously added on
Vital-IT, see <span style='color:blue'>blast_gene_families.sh</span>). RefSeq
database is suitable as it is big (and manually curated) and it will allow us
to access taxonomic informations from NCBI for further ‘Phylogeny’ step. <span
style='color:blue'>blast_gene_families.sh</span> runs blastp command specifying
queries and outputs. <span style='color:blue'>function_blast.sh</span> creates
output folders (<span style='color:#980000'>data/blast/Bumble_bees_proteins,
data/blast/Honey_bees_proteins, data/blast/Bumble_Honey_bees_proteins</span>)
and assigns variables used in <span style='color:blue'>blast_gene_families.sh</span>
(variables correspond to every <span style='color:#38761D'>Gene_family_*.fasta</span>
files present in each folder of group bees and stored in <span
style='color:#38761D'>list_files.txt</span>).</p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Inputs: outputs of </span><span style='color:blue'>proteinseqextract.py</span>
- <span style='color:#38761D'>Gene_family_*.fasta</span> in <span
style='color:#980000'>data/Bumble_bees_proteins, data/Honey_bees_proteins,
data/Bumble_Honey_bees_proteins</span> (to select <span style='color:#38761D'>Gene_family_*.fasta</span>,
<span style='color:blue'>function_blast.sh</span> will go through <span
style='color:#38761D'>list_files.txt</span> - created in the terminal: <b
style='mso-bidi-font-weight:normal'>ls Gene_family_* &gt; list_files.txt</b>)<o:p></o:p></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Outputs: Gene_family_*.out </span>of its corresponding <span
style='color:#38761D'>Gene_family_*.fasta</span> of its corresponding folder of
group of bees <span style='color:#980000'>(data/Bumble_bees_proteins,
data/Honey_bees_proteins, data/Bumble_Honey_bees_proteins)</span> in its
corresponding output folders <span style='color:#980000'>(data/blast/Bumble_bees_proteins,
data/blast/Honey_bees_proteins, data/blast/Bumble_Honey_bees_proteins)</span> -
each <span style='color:#38761D'>Gene_family_*.out</span> file contains blast
hits for each reference_genome|protein_ID’s sequence - for each
reference_genome|protein_ID’s sequence (query): 5 first lines summarizing
blastp information (version of BLASTP, query, database, fields and the total
number of found hits) - fields correspond to the name of each resulting column
(total of 14 columns) - subsequent lines correspond to every hits of the query</i></p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'>4. For every blast results of each
reference_genome|protein_ID’s sequence (query) present in each <span
style='color:#38761D'>Gene_family_*.out</span> files in each folder of group of
bees (<span style='color:#980000'>data/blast/Bumble_bees_proteins,
data/blast/Honey_bees_proteins, data/blast/Bumble_Honey_bees_proteins</span>), <span
style='color:blue'>blast_hits_extract.py</span> parses hits according to
different parameters (threshold_alignment_length, threshold_ID, threshold_eval,
threshold_bitscore): </p>

<p class=normalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-align:justify;text-indent:-18.0pt;mso-list:l1 level1 lfo1'><![if !supportLists]><span
style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><![endif]>threshold_alignment_length : constant = 0.8</p>

<p class=normalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-align:justify;text-indent:-18.0pt;mso-list:l1 level1 lfo1'><![if !supportLists]><span
style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><![endif]>threshold_ID : starts at 50</p>

<p class=normalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-align:justify;text-indent:-18.0pt;mso-list:l1 level1 lfo1'><![if !supportLists]><span
style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><![endif]>threshold _eval : 0.00001</p>

<p class=normalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-align:justify;text-indent:-18.0pt;mso-list:l1 level1 lfo1'><![if !supportLists]><span
style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><![endif]>threshold_bitscore : 0</p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'>For a given <span style='color:#38761D'>Gene_family_*.out</span>,
if there are less than 25 blast hits with parameters below, we relax the
threshold_ID by multiplying it by 0.8 until reaching at least 25 blast hits. <span
style='color:blue'>blast_hits_extract.sh</span> creates output folders (<span
style='color:#980000'>data/parsed_blast/Bumble_bees_proteins,
data/parsed_blast/Honey_bees_proteins,
data/parsed_blast/Bumble_Honey_bees_proteins</span>) and assigns parameters as
below. </p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Inputs: outputs of </span><span style='color:blue'>blast_gene_families.sh</span>
and <span style='color:blue'>function_blast.sh</span> - <span style='color:
#38761D'>Gene_family_*.out</span> in <span style='color:#980000'>data/blast/Bumble_bees_proteins,
data/blast/Honey_bees_proteins, data/blast/Bumble_Honey_bees_proteins</span>
(to select <span style='color:#38761D'>Gene_family_*.out</span>, <span
style='color:blue'>blast_hits_extract.py</span> will go through <span
style='color:#38761D'>list_files.txt </span>- created in the terminal: <b
style='mso-bidi-font-weight:normal'>ls Gene_family_* &gt; list_files.txt</b>)<o:p></o:p></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Outputs: Gene_family_*_parsed.out </span>of its
corresponding <span style='color:#38761D'>Gene_family_*.out</span> of its
corresponding folder of group of bees <span style='color:#980000'>(data/blast/Bumble_bees_proteins,
data/blast/Honey_bees_proteins, data/blast/Bumble_Honey_bees_proteins)</span>
in its corresponding output folders <span style='color:#980000'>(data/parsed_blast/Bumble_bees_proteins,
data/parsed_blast/Honey_bees_proteins,
data/parsed_blast/Bumble_Honey_bees_proteins)</span> - each <span
style='color:#38761D'>Gene_family_*_parsed.out</span> file contains the ‘best’
blast hits for each reference_genome|protein_ID’s sequence - the header is ‘#
Query_ID<span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Subject_titles<span
style='mso-tab-count:1'> </span>%_Identity<span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Alignment_length<span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>evalue<span
style='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>bit_score’
- for each ‘best’ hits, informations are stored in its corresponding column<o:p></o:p></i></p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'>5. From every <b style='mso-bidi-font-weight:
normal'>parsed</b> blast results of each reference_genome|protein_ID’s sequence
(query) present in each <span style='color:#38761D'>Gene_family_*_parsed.out</span>
files in each folder of group of bees (<span style='color:#980000'>data/parsed_blast/Bumble_bees_proteins,
data/parsed_blast/Honey_bees_proteins, data/parsed_blast/Bumble_Honey_bees_proteins</span>),
<span style='color:blue'>extract_taxonomy_hierarchy.py</span> first creates a
list of subject IDs (hit IDs, for example ‘WP_052726720’). Subsequently, thanks
to Entrez (from Bio package), the function extracts the hierarchical taxonomy
of each subject IDs (from protein database of NCBI, for example:
'Bacteria;Firmicutes;Bacilli;Lactobacillales;Lactobacillaceae;Lactobacillus'). <i
style='mso-bidi-font-style:normal'>The example (‘WP_052726720’) is used as
reference hierarchical taxonomy.</i> Finally, the function looks for the most
recent taxa which is shared between each hierarchical taxonomy of each subject
IDs and the reference hierarchical taxonomy. Hierarchical taxonomy distance are
subjective:</p>

<p class=normalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-align:justify;text-indent:-18.0pt;mso-list:l0 level1 lfo2'><![if !supportLists]><span
style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><![endif]>Lactobacillus = 1</p>

<p class=normalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-align:justify;text-indent:-18.0pt;mso-list:l0 level1 lfo2'><![if !supportLists]><span
style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><![endif]>Lactobacillaceae = 2</p>

<p class=normalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-align:justify;text-indent:-18.0pt;mso-list:l0 level1 lfo2'><![if !supportLists]><span
style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><![endif]>Lactobacillales = 3</p>

<p class=normalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-align:justify;text-indent:-18.0pt;mso-list:l0 level1 lfo2'><![if !supportLists]><span
style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><![endif]>Bacilli = 4</p>

<p class=normalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-align:justify;text-indent:-18.0pt;mso-list:l0 level1 lfo2'><![if !supportLists]><span
style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><![endif]>Firmicutes = 5</p>

<p class=normalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-align:justify;text-indent:-18.0pt;mso-list:l0 level1 lfo2'><![if !supportLists]><span
style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><![endif]>Bacteria = 6</p>

<p class=normalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-align:justify;text-indent:-18.0pt;mso-list:l0 level1 lfo2'><![if !supportLists]><span
style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span><![endif]>None = 7 (which either corresponds to Archae or
Eukaryota - contaminations)</p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'>The function creates a summary file
(<span style='color:#38761D'>hierarchical_taxonomy.txt</span>) which is located
in <span style='color:#980000'>data/parsed_blast</span>.</p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Inputs: Gene_family_*_parsed.out </span>of its
corresponding folder of group of bees <span style='color:#980000'>(data/parsed_blast/Bumble_bees_proteins,
data/parsed_blast/Honey_bees_proteins,
data/parsed_blast/Bumble_Honey_bees_proteins)</span> <o:p></o:p></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Outputs: hierarchical_taxonomy.txt</span> in <span
style='color:#980000'>data/parsed_blast </span>- the first column corresponds
to the strain IDs, the second one corresponds to the hierarchical taxonomy and
the third one corresponds to the subjective hierarchical taxonomy distance<o:p></o:p></i></p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'>6. For every <b style='mso-bidi-font-weight:
normal'>parsed</b> blast results of each reference_genome|protein_ID’s sequence
(query) present in each <span style='color:#38761D'>Gene_family_*_parsed.out</span>
files in each folder of group of bees (<span style='color:#980000'>data/parsed_blast/Bumble_bees_proteins,
data/parsed_blast/Honey_bees_proteins,
data/parsed_blast/Bumble_Honey_bees_proteins</span>), <span style='color:blue'>plot_identity_taxonomy.R</span>
extracts its percentage of identity and its hierarchical taxonomy distance (in <span
style='color:#38761D'>hierarchical_taxonomy.txt</span>) and plot them against
each other. Moreover, for each <span style='color:#38761D'>Gene_family_*_parsed.out</span>,
on one hand, we will perform linear regression model in addition to polynomial
model and investigate if models differ. If they differ, it would mean that
polynomial model is the best one and suggest potential horizontal gene
transfer. </p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Inputs: Gene_family_*_parsed.out </span>of its
corresponding folder of group of bees <span style='color:#980000'>(data/parsed_blast/Bumble_bees_proteins,
data/parsed_blast/Honey_bees_proteins, data/parsed_blast/Bumble_Honey_bees_proteins)</span>
and <span style='color:#38761D'>hierarchical_taxonomy.txt</span> in <span
style='color:#980000'>data/parsed_blast</span><o:p></o:p></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Outputs: Bumble_taxo_plot.pdf, Honey_taxo_plot.pdf,
Bumble_Honey_taxo_plot.pdf</span> and <span style='color:#38761D'>potential_HGT.txt</span>
in <span style='color:#980000'>data/parsed_blast</span> - each <span
style='color:#38761D'>*.pdf</span> file contains all plots of a given bee group
- <span style='color:#38761D'>potential_HGT.txt</span> contains a list of <span
style='color:#38761D'>Gene_family_*_parsed.out</span> that are orthologous
genes potentially acquired by HGT<o:p></o:p></i></p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'>7. Manually, we decide which gene
families are good candidates to investigate if orthologous genes are acquired
by HGT. These gene family candidates are listed in <span style='color:#38761D'>list_files_HGT_to_trees.txt</span>
(in <span style='color:#980000'>data/parsed_blast</span>). <span
style='color:#38761D'>Gene_family_1058_parsed.out</span> and <span
style='color:#38761D'>Gene_family_991_parsed.out</span> (<span
style='color:#980000'>data/parsed_blast/Bumble_Honey_bees_proteins</span>)
correspond to ‘confirmed’ HGT candidates. <span style='color:#38761D'>Gene_family_1099_parsed.out</span>
(<span style='color:#980000'>data/parsed_blast/Bumble_Honey_bees_proteins</span>)
and <span style='color:#38761D'>Gene_family_1674_parsed.out </span>(<span
style='color:#980000'>data/parsed_blast/Bumble_bees_proteins</span>) correspond
to ‘supposed’ HGT candidates. <span style='color:#38761D'>Gene_family_1048_parsed.out</span>
(<span style='color:#980000'>data/parsed_blast/Bumble_Honey_bees_proteins</span>)
corresponds to non-HGT candidate. First, <span style='color:blue'>extract_seq_blast_hits.py</span>
extracts the protein sequence of every blast hits of selected <span
style='color:#38761D'>Gene_family_*_parsed.out</span> (listed in <span
style='color:#38761D'>list_files.txt</span>). From this, <span
style='color:blue'>align_potential_HGT.sh</span> aligns the sequence of every
blast hits of selected <span style='color:#38761D'>Gene_family_*_parsed.out</span>
(listed in <span style='color:#38761D'>list_files.txt</span>) in its folder of
group of bees (<span style='color:#980000'>data/phylogeny_potential_HGT/Bumble_bees_proteins,
data/phylogeny_potential_HGT/Honey_bees_proteins,
data/phylogeny_potential_HGT/Bumble_Honey_bees_proteins</span>). Then, <span
style='color:blue'>tree_wo_bootstrap.sh</span> runs RAXML tool to infer
phylogenetic trees (without bootstrapping) of all aligned protein sequences of
every blast hits of selected <span style='color:#38761D'>Gene_family_*_parsed.out</span><span
style="mso-spacerun:yes">&nbsp; </span>(in its bee folder in <span
style='color:#980000'>data/phylogeny_potential_HGT</span>). </p>

<p class=normal style='text-align:justify'><span style='color:blue'>extract_seq_blast_hits.py</span>:<i
style='mso-bidi-font-style:normal'><o:p></o:p></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Inputs: list_files.txt</span> in <span style='color:#980000'>data/parsed_blast</span>,
<span style='color:#38761D'>Gene_family_*_parsed.out </span>of its
corresponding folder of group of bees <span style='color:#980000'>(data/parsed_blast/Bumble_bees_proteins,
data/parsed_blast/Honey_bees_proteins,
data/parsed_blast/Bumble_Honey_bees_proteins) </span>listed in <span
style='color:#38761D'>list_files.txt</span><o:p></o:p></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Outputs: amino_acid_seq_Gene_family_*_parsed.fasta </span>in
<span style='color:#980000'>data/parsed_blast</span> - each output file
contains protein sequences of every blast hits of a selected gene family. The
ID name of each sequence is defined as follows:
StrainName_SujectID_HierarchicalTaxonomyDistance</i></p>

<p class=normal style='text-align:justify'><span style='color:blue'>align_potential_HGT.sh</span>:<i
style='mso-bidi-font-style:normal'><o:p></o:p></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Inputs: amino_acid_seq_Gene_family_*_parsed.fasta </span>in
<span style='color:#980000'>data/parsed_blast</span><span style='color:#38761D'><o:p></o:p></span></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Outputs: Gene_family_*_parsed.multifasta</span> in its
corresponding folder of group of bees <span style='color:#980000'>(data/phylogeny_potential_HGT/Bumble_bees_proteins,
data/phylogeny_potential_HGT/Honey_bees_proteins,
data/phylogeny_potential_HGT/Bumble_Honey_bees_proteins)</span> - each output
file contains aligned protein sequences of every blast hits of a selected gene
family<o:p></o:p></i></p>

<p class=normal style='text-align:justify'><span style='color:blue'>tree_wo_bootstrap.sh</span>:<i
style='mso-bidi-font-style:normal'><o:p></o:p></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Inputs: Gene_family_*_parsed.multifasta</span> in its
corresponding folder of group of bees <span style='color:#980000'>(data/phylogeny_potential_HGT/Bumble_bees_proteins,
data/phylogeny_potential_HGT/Honey_bees_proteins,
data/phylogeny_potential_HGT/Bumble_Honey_bees_proteins)<o:p></o:p></span></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Outputs: *ML_wo_bootstrap_Gene_family_*_parsed.out </span>in
<span style='color:#980000'>data/phylogeny_potential_HGT/RAxML_results<o:p></o:p></span></i></p>

<p class=normal style='text-align:justify'>&gt;&gt;
RAxML_bestTree.ML_wo_bootstrap_Gene_family_*_parsed.out can then be opened
using FigTree tool (available at http://tree.bio.ed.ac.uk/software/figtree/)</p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

<p class=normal style='text-align:justify'>8. The inference of genomic islands
is performed using Islandviewer tool (available at
http://www.pathogenomics.sfu.ca/islandviewer/). For this purpose, two reference
genomes from Bumble bees group (<b style='mso-bidi-font-weight:normal'><i
style='mso-bidi-font-style:normal'>F5_237</i></b> and <b style='mso-bidi-font-weight:
normal'><i style='mso-bidi-font-style:normal'>F5_245</i></b>) and one from
Honey bees group (<b style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:
normal'>Lb_183</i></b>) are analysed. For each referenge genome, <span
style='color:blue'>extract_coordinates_function_from_gbkfiles.py</span>
extracts the protein name of every genes present in each Gene_family_* from <span
style='color:#38761D'>GeneFamilies.txt </span>(in <span style='color:#990000'>data</span>).
Thanks to GeneBank files (<span style='color:#990000'>mls_2016/genome_files</span>),
the coordinates and the function of each protein are recovered. Then, we map
manually the specific position of the protein onto its reference genome.
Results are present in <span style='color:#980000'>data/IslandViewerResults, </span><span
style='color:#38761D'>*.png</span><span style='color:#00B050'> </span>shows the
circular representation of each reference genome and the position of the
genomic islands,<span style="mso-spacerun:yes">&nbsp; </span><span
style='color:#38761D'>*.tsv</span><span style='color:#00B050'> </span>contains
the coordinates of each genomic island and which method allows this inference.
Moreover, <span style='color:#38761D'>MappingIntoGenome.txt</span><span
style='color:#00B050'> </span>sums up the different results for this part.</p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Input: GeneFamilies.txt</span> in <span style='color:
#980000'>data</span>, <span style='color:#38761D'>GeneBank files</span> in <span
style='color:#990000'>mls_2016/genome_files</span><o:p></o:p></i></p>

<p class=normal style='text-align:justify'><i style='mso-bidi-font-style:normal'><span
style='color:#38761D'>Outputs: *.png files</span> , <span style='color:#38761D'>*.tsv
files </span>and<span style='color:#38761D'> MappingIntoGenome.txt </span>in<span
style='color:#38761D'> </span><span style='color:#980000'>data/IslandViewerResults</span></i></p>

<p class=normal style='text-align:justify'><o:p>&nbsp;</o:p></p>

</div>

</body>

</html>
